tests/protocols/test_http.py

```diff
@@ -123,9 +123,10 @@ def set_protocol(self, protocol):
 
```

Comment:
the whole diff may seem hugs, but it is simple: the MockLoop takes a parameter event_llop, wich is the pytest-asyncio fixture, we get nice cleanup with it. 
So in order to use that, each test is decorated with @pytest.mark.asyncio
And the `get_connected_protocol` function is tunred into a context manager so that we can also clean it on every use